### PR TITLE
3817 - schema adjustments for event list

### PR DESCRIPTION
**Pull request by chiaberry**
# Description

This is the joplin part of the Events List view on Janis. I needed to adjust the query so events would be a) return after a certain date, and b) in date order. I set it up so we can also return events before a certain date and the reverse date order, for when we eventually show past events. 

```
query getEventPage($id: ID, $date_Lte: Date, $date_Gte: Date) {
 	allEventPages(orderBy:"date", id:$id, date_Lte:$date_Lte, date_Gte:$date_Gte) {
    edges {
      node { }
    }
  }
}
```

adjust the `getEventPage` query by adding `orderBy: "date"` for ascending order or `orderBy: "-date"` for descending. 

date_Lte returns dates "less than or equal to" and Gte is greater than or equal to. Date are sent in "2020-03-02" format.

https://django-filter.readthedocs.io/en/master/ref/filterset.html#declaring-filterable-fields
https://docs.djangoproject.com/en/3.0/ref/models/querysets/#lte
https://github.com/graphql-python/graphene-django/blob/master/examples/cookbook/cookbook/ingredients/schema.py

# Janis instance linked to this
https://janis-3817-event-lists.netlify.com/events/

# Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How can this be tested?

 

# Checklist:
- [ ] Request reviewers
- [ ] Slack-out message for review
- [ ] Link this PR in the issue
- [ ] Moved card to *review* in zenhub

### CODE REVIEW

File: joplin/api/schema.py [hunk 1]
```diff
@@ -231,6 +231,22 @@ class Meta:
...
+    order_by = OrderingFilter(
+        fields=(
+            ('date'),
+        )
+    )
+
+    class Meta:
+        model = EventPage
+        fields = {
+            'date': ['exact', 'lte', 'gte'],
```
Comment (easherma) [inline joplin/api/schema.py]: I think we talked about this, but adding a note/link for more info on where 'lte' and 'gte' come from, etc. so we know where to dig for more info

File: joplin/api/schema.py [hunk 2]
```diff
@@ -231,6 +231,22 @@ class Meta:
         interfaces = [graphene.Node]
 
 
+class EventFilter(FilterSet):
```
Comment (easherma) [inline joplin/api/schema.py]: Can we see an example of where this is used? 
This is probs a stretch goal (and maybe we can make a separate issue), but we should have a test that this ordering filter works correctly if we're going to  roll our own schema ( or we can use https://github.com/linuxsoftware/ls.joyous which already has test coverage )

File: joplin/api/schema.py (hunk 1 above)
Comment (chiaberry) [reply joplin/api/schema.py]: I put it in the PR, do you think it should be also as a comment in the code?

File: joplin/api/schema.py (hunk 2 above)
Comment (chiaberry) [reply joplin/api/schema.py]: example of where it's used on Janis? getAllEvents in static.config

Comment (easherma) [reply joplin/api/schema.py]: here's an example of a simple API test case btw: 
https://github.com/cityofaustin/joplin/pull/545/files

File: joplin/api/schema.py (hunk 1 above)
Comment (easherma) [reply joplin/api/schema.py]: I personally wouldn't be opposed to a link to https://docs.djangoproject.com/en/3.0/ref/models/querysets/#field-lookups
, but a least a comment like 
'these 'fields' correspond to django queryset field-lookups' so future us knows where to look (and also where to _not_ look ) in docs

**Pull request merged**